staging/src/k8s.io/client-go/transport/config.go

```diff
@@ -99,4 +105,6 @@ type TLSConfig struct {
 	CAData   []byte // Bytes of the PEM-encoded server trusted root certificates. Supercedes CAFile.
 	CertData []byte // Bytes of the PEM-encoded client certificate. Supercedes CertFile.
 	KeyData  []byte // Bytes of the PEM-encoded client key. Supercedes KeyFile.
+
+	GetCert func() (*tls.Certificate, error) // Callback that returns a TLS client certificate. Supercedes CertData, CertFile, KeyData and KeyFile.
```

Comment:
@kubernetes/sig-api-machinery-pr-reviews this makes transport.TLSConfig non-serializable. From what I can see, it is only used in transport.Config, which is also non-serializable (has WrapTransport, Dial, etc), so I don't think that's a problem, but wanted to call it out

Reply:
the comment about "supercedes certdata, ..." isn't accurate any more. see `Note: static key/cert data always take precedence over cert callback.` comment below

Reply:
Updated comment